src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAITextToImageClient.cs

```diff
@@ -0,0 +1,252 @@
+    {
+        _ = Throw.IfNull(prompt);
+        _ = Throw.IfNull(options);
+
+        ImageGenerationOptions openAIOptions = ToOpenAIImageGenerationOptions(options);
+        ImageClient imageClient = GetImageClient(options);
+
+        GeneratedImageCollection result = await imageClient.GenerateImagesAsync(prompt, options.Count, openAIOptions, cancellationToken).ConfigureAwait(false);
+
+        return ToTextToImageResponse(result, options);
+    }
+
+    /// <inheritdoc />
+    public async Task<TextToImageResponse> GenerateEditImageAsync(
+        Stream originalImage, string originalImageFileName, string prompt, TextToImageOptions? options, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(originalImage);
+        _ = Throw.IfNull(originalImageFileName);
+        _ = Throw.IfNull(prompt);
+        _ = Throw.IfNull(options);
+
+        ImageEditOptions openAIOptions = ToOpenAIImageEditOptions(options);
+        ImageClient imageClient = GetImageClient(options);
+
+        GeneratedImageCollection result = await imageClient.GenerateImageEditsAsync(
+            originalImage, originalImageFileName, prompt, options.Count, openAIOptions, cancellationToken).ConfigureAwait(false);
+
+        return ToTextToImageResponse(result, options);
+    }
+
+    /// <inheritdoc />
+#pragma warning disable S1067 // Expressions should not be too complex
+    public object? GetService(Type serviceType, object? serviceKey = null) =>
+        serviceType is null ? throw new ArgumentNullException(nameof(serviceType)) :
+        serviceKey is not null ? null :
+        serviceType == typeof(TextToImageClientMetadata) ? _metadata :
+        serviceType == typeof(ImageClient) ? _imageClient :
+        serviceType == typeof(OpenAIClient) ? _openAIClient :
+        serviceType.IsInstanceOfType(this) ? this :
+        null;
+#pragma warning restore S1067 // Expressions should not be too complex
+
+    /// <inheritdoc />
+    void IDisposable.Dispose()
+    {
+        // Nothing to dispose. Implementation required for the ITextToImageClient interface.
+    }
+
+    /// <summary>Converts a <see cref="GeneratedImageCollection"/> to a <see cref="TextToImageResponse"/>.</summary>
+    private static TextToImageResponse ToTextToImageResponse(GeneratedImageCollection generatedImages, TextToImageOptions options)
+    {
+        List<AIContent> contents = new();
+
+        foreach (GeneratedImage image in generatedImages)
+        {
+            if (options.ContentType == TextToImageContentType.Uri && image.ImageUri is not null)
+            {
+                contents.Add(new UriContent(image.ImageUri, "image/png"));
+            }
+            else if (options.ContentType == TextToImageContentType.Data && image.ImageBytes is not null)
+            {
+                contents.Add(new DataContent(image.ImageBytes.ToArray(), "image/png"));
```

Comment:
Are they always png? It doesn't send back a media/mime type or anything?

Reply:
I was able to workaround it using private reflection.